### PR TITLE
:sparkles: Documentation for checks active, frozen-deps, SAST and security policy

### PR DESCRIPTION
**Pull request by naveensrinivasan**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
N/A

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
 #304

* **What is the current behavior?** (You can also link to an open issue here)
Does not provide docs

* **What is the new behavior (if this is a feature change)?**
 Included docs in the results for active, deps,sast and security_policy

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
None

* **Other information**:
None

### CODE REVIEW

File: checks/active.go [hunk 1]
```diff
@@ -51,5 +51,8 @@ func IsActive(c checker.Checker) checker.CheckResult {
 	return checker.CheckResult{
 		Pass:       totalCommits >= numCommits,
 		Confidence: confidence,
+		Description: `This check tries to determine if the project is still "actively maintained".
```
Comment (inferno-chromium) [inline checks/active.go:54]: i dont like the idea of adding description here, help url should be enough and we shouldnt duplicate the doc text which can get updated with time (and also will be detailed).

File: checks/active.go [hunk 2]
```diff
@@ -51,5 +51,8 @@ func IsActive(c checker.Checker) checker.CheckResult {
 	return checker.CheckResult{
 		Pass:       totalCommits >= numCommits,
 		Confidence: confidence,
+		Description: `This check tries to determine if the project is still "actively maintained". 
+		It currently works by looking for commits within the last 90 days.`,
+		HelpURL: "https://github.com/ossf/scorecard/blob/main/checks.md#active",
```
Comment (inferno-chromium) [inline checks/active.go:56]: we should move https://github.com/ossf/scorecard/blob/main/checks.md into a constant in check.go, and this file can just pass the #<marker-name>

File: checks/sast.go [hunk 3]
```diff
@@ -47,12 +56,19 @@ func SASTToolInCheckRuns(c checker.Checker) checker.CheckResult {
 			continue
 		}
 		totalMerged++
-		crs, _, err := c.Client.Checks.ListCheckRunsForRef(c.Ctx, c.Owner, c.Repo, pr.GetHead().GetSHA(), &github.ListCheckRunsOptions{})
+		crs, _, err := c.Client.Checks.
+			ListCheckRunsForRef(c.Ctx, c.Owner, c.Repo, pr.GetHead().GetSHA(), &github.ListCheckRunsOptions{})
 		if err != nil {
-			return checker.RetryResult(err)
+			r := checker.RetryResult(err)
+			r.Description = description
```
Comment (inferno-chromium) [inline checks/sast.go:63]: helper function, rather than repeating this.

**Review by inferno-chromium — CHANGES_REQUESTED** [review]

File: checks/active.go (hunk 1 above)
Comment (naveensrinivasan) [reply checks/active.go:54]: For `sarif` description is required.SARIF is a standard format and is being used at the GitHub Security center. How do we deal with that?

Comment (inferno-chromium) [reply checks/active.go:54]: Then need to move all strings data in xml file and then fetch description from there. we can have two fields, summary and detailed description. summary shows by default. we shouldn't put these long strings in code.

Comment (naveensrinivasan) [reply checks/active.go:54]: image

Which is the summary? The above section?

Comment (inferno-chromium) [reply checks/active.go:54]: Basically i want https://github.com/ossf/scorecard/blob/main/checks.md to be converted into a machine readable format, with summary, description and remediation fields. And then we read them in code and show it (normal view-summary only or detailed view which shows description and remediation as well). i want to reduce deduplication of text in checks.md and also dont want to hardcode strings in source code.

Comment (naveensrinivasan) [reply checks/active.go:54]: OK. I understand. But the issue is that if we were covert it to machine-readable then how do we provide the same information as markdown when someone visits the GitHub Repo. Ideally, we have a `web` frontend that can serve this data and can be the single source of truth.

Comment (inferno-chromium) [reply checks/active.go:54]: I am fine to remove checks.md for now. We can later add it, can you create an issue.

Comment (naveensrinivasan) [reply checks/active.go:54]: OK. I am fine with that. Just an FYI... this is a chore and would take some time.

Comment (naveensrinivasan) [reply checks/active.go:54]: This is required to get the SARIF results and only after that, we can publish the scorecard as a GitHub action.

Comment (naveensrinivasan) [reply checks/active.go:54]: We need more contributors.

Comment (inferno-chromium) [reply checks/active.go:54]: We have a new member starting on Scorecard project, so we should get help soon. Can table this one for now.

Comment (naveensrinivasan) [reply checks/active.go:54]: Sounds good. Thanks

Comment (oliverchang) [reply checks/active.go:54]: big +1 to machine readable format for checker descriptions. We need this info for consumers of scorecard data to show in their own UI.

**Comment by naveensrinivasan** [comment]
Closing this PR to implement it based on the latest docs #345

**Pull request closed without merging**